tao_compiler/mlir/disc/transforms/disc_shape_simplifier.cc

```diff
@@ -202,7 +202,12 @@ class DynamicReshapeOpShapeInference
   }
 };
 
-// Match following patterns:
+// Convert an expand-like reshape op followed by a dynamic-broadcast-in-dim op
+// to a single dynamic-broadcast-in-dim op. The dim size of expanded dims should
+// be one.
+//
+// An example is like following:
```

Comment:
From the code, we are looking for a DynamicReshapeOp or ReshapeOp to match the pattern. But i didn't see any reshape here in the example?

Reply:
This example only shows a dynamic-reshape based pattern. ReshapeOp based pattern is similar except that the batch-dim value is constant folded.